middleware/fetchUser.js

```diff
@@ -0,0 +1,26 @@
+const joi = require('joi');
+const logger = require('../utils/logger');
+
+const validateFetchUser = async (req, res, next) => {
+  const schema = joi.object({
+    first_name: joi.string(),
+    last_name: joi.string(),
+    email: joi.string(),
+    phone: joi.string(),
+    country: joi.string(),
+    state: joi.string(),
+    zip_code: joi.string(),
+  });
+
+  try {
+    await schema.validateAsync( req.body, req.params.id);
```

Comment:
This is wrong, you should only write the validation schema for the params id and not the whole user object. We only need to validate the payload we receive upon each request, whether it's a route param, a route query or a request body.